sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -274,6 +277,34 @@ private String getAndValidateFullyQualifiedNamespace() {
         return fullyQualifiedNamespace;
     }
 
+    /**
+     * Sets a custom endpoint address when connecting to the Service Bus service. This can be useful when your network
+     * does not allow connecting to the standard Azure Service Bus endpoint address, but does allow connecting through
+     * an intermediary. For example: {@literal https://my.custom.endpoint.com:55300}.
+     * <p>
+     * If no port is specified, the default port for the {@link #transportType(AmqpTransportType) transport type} is
+     * used.
+     *
+     * @param customEndpointAddress The custom endpoint address.
+     * @return The updated {@link ServiceBusClientBuilder} object.
+     * @throws IllegalArgumentException if {@code customEndpointAddress} cannot be parsed into a valid {@link URL}.
+     */
+    public ServiceBusClientBuilder customEndpointAddress(String customEndpointAddress) {
+        if (customEndpointAddress == null) {
```

Comment:
For Event Hubs, It seems we support setting custom endpoint at Processor level as well (in addition to EventHubsClientBuilder) - https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/eventhubs/azure-messaging-eventhubs/CHANGELOG.md#550-2020-02-15 Should we check if its the same for SB processor as well

Reply:
Yeah, SB processor client is came from  `ServiceBusClientBuilder` and different from EH processor client, we don't need to add new method in both `EventHubClientBuilder` and `EventProcessorClientBuilder` like EH does. All of the service bus client only use the same method `ServiceBusClientBuilder.getOrCreateConnectionProcessor` which use custom endpoint. Overall current code for custom endpoint works for Processor level.
//cc @conniey

Reply:
Thanks for checking Zeija.. lgtm then!